### ISSUE TITLE
Key 0x0115 not found

### ISSUE THREAD

**Issue by jowch** [tlnagy/TiffImages.jl#56]
Hello! I am hoping to use your package to analyze some images. It seems that there is a metadata tag in my image that isn't recognized (0x0115). My images are generated with some old software and I am no expert at TIFF. Seems that this tag denotes ["SamplesPerPixel"](https://www.awaresystems.be/imaging/tiff/tifftags/samplesperpixel.html). I followed some of the suggestions on #23 to debug the issue:

```Julia
julia> img = TiffImages.load("example.tif")
ERROR: KeyError: key 0x0115 not found

julia> tf = read(open(fpath), TiffImages.TiffFile)
TiffFile{UInt32, FileIO.Stream{FileIO.DataFormat{:TIFF}, IOStream, String}}("example.tif", FileIO.Stream{FileIO.DataFormat{:TIFF}, IOStream, String}(IOStream(<file example.tif>), "example.tif"), 130, false)

julia> first(tf)
IFD, with tags:
	Tag(SUBFILETYPE, 0)
	Tag(IMAGEWIDTH, 981)
	Tag(IMAGELENGTH, 1043)
	Tag(BITSPERSAMPLE, 32)
	Tag(COMPRESSION, COMPRESSION_NONE)
	Tag(PHOTOMETRIC, 1)
	Tag(IMAGEDESCRIPTION, REMOTE@580 String[] len=508)
	Tag(MODEL, REMOTE@62 String[] len=48)
	Tag(STRIPOFFSETS, 4096)
	Tag(ROWSPERSTRIP, 1043)
	Tag(STRIPBYTECOUNTS, 4092732)
	Tag(XRESOLUTION, REMOTE@484 Rational{UInt32}[] len=1)
	Tag(YRESOLUTION, REMOTE@492 Rational{UInt32}[] len=1)
	Tag(SOFTWARE, REMOTE@110 String[] len=20)
	Tag(DATETIME, REMOTE@30 String[] len=32)
	Tag(ARTIST, REMOTE@10 String[] len=20)
	Tag(SAMPLEFORMAT, 2)
	Tag(EXIFTAG_EXIFVERSION, "�7t�")
	Tag(UNKNOWN(36865), 0)
	Tag(UNKNOWN(36866), 0)
	Tag(EXIFTAG_DATETIMEORIGINAL, 1.8)
	Tag(EXIFTAG_DATETIMEDIGITIZED, 0.0)
	Tag(UNKNOWN(36870), "")
	Tag(UNKNOWN(36876), 0.0)
	Tag(UNKNOWN(36877), 0.0)
	Tag(UNKNOWN(36878), 0.0)
	Tag(UNKNOWN(36879), 0.0)
	Tag(UNKNOWN(36880), 0.0)
	Tag(UNKNOWN(37120), REMOTE@500 UInt32[] len=20)
```

I've included an example image here to help with diagnosis. I zipped it as I can't upload tif files here. I can send it in an alternative way if needed. Thanks!
[example.tif.zip](https://github.com/tlnagy/TiffImages.jl/files/6727746/example.tif.zip)

**Comment by tlnagy** [comment]
Thanks for all the details, it definitely looks like your image is missing the `SamplesPerPixel` tag. It appears that `TiffImages.jl` should assume a value of 1 in that case.

**Comment by tlnagy** [comment]
Now that #57 is merged could you pull the latest master by running 

```julia
using Pkg
Pkg.develop("TiffImages")
```

in your REPL. And see if that fixes your problem?

Also, it looks like your image is used a signed integer data type, support for this is [a little clunky](https://tamasnagy.com/TiffImages.jl/dev/examples/writing/#Signed-integers) in Julia's display libraries. For display, you use reinterpret in the following way:

```julia
using FixedPointNumbers, Colors
img = TiffImages.load("example.tif")
reinterpret(Gray{N0f32}, img)
```

**Comment by jowch** [comment]
Thanks a lot for your help! It seems to work without error now.

**Comment by tlnagy** [comment]
Great! I've released a new stable version (v0.4.1) with this bugfix. I recommend switching your `TiffImages.jl` back to stable by running

```
] free TiffImages
```

in the REPL.
